android/java/org/chromium/chrome/browser/util/BraveTouchUtils.java

```diff
@@ -0,0 +1,100 @@
+     *
+     * Unless required by applicable law or agreed to in writing, software
+     * distributed under the License is distributed on an "AS IS" BASIS,
+     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     * See the License for the specific language governing permissions and
+     * limitations under the License.
+     */
+    public static void ensureMinTouchTarget(View view, int minTarget) {
+        View parent = (View) view.getParent();
+        if (parent != null) {
+            parent.getViewTreeObserver().addOnGlobalLayoutListener(new OnGlobalLayoutListener() {
+                @Override
+                public void onGlobalLayout() {
+                    if (view.isShown()) {
+                        final Rect bounds = new Rect();
+                        view.getHitRect(bounds);
+
+                        DisplayMetrics metrics =
+                                view.getContext().getResources().getDisplayMetrics();
+                        int height = (int) Math.round(bounds.height() / metrics.density);
+                        int width = (int) Math.round(bounds.width() / metrics.density);
+
+                        int extraSpaceStart = 0;
+                        int extraSpaceEnd = 0;
+                        if (height < minTarget) {
+                            extraSpaceStart = (int) Math.floor((minTarget - height) / 2f);
+                            extraSpaceEnd = minTarget - height - extraSpaceStart;
+                            extraSpaceStart = (int) Math.ceil(extraSpaceStart * metrics.density);
+                            extraSpaceEnd = (int) Math.ceil(extraSpaceEnd * metrics.density);
+                            bounds.top -= extraSpaceStart;
+                            bounds.bottom += extraSpaceEnd;
+                        }
+
+                        if (width < minTarget) {
+                            extraSpaceStart = (int) Math.floor((minTarget - width) / 2f);
+                            extraSpaceEnd = minTarget - width - extraSpaceStart;
+                            extraSpaceStart = (int) Math.ceil(extraSpaceStart * metrics.density);
+                            extraSpaceEnd = (int) Math.ceil(extraSpaceEnd * metrics.density);
+                            bounds.left -= extraSpaceStart;
+                            bounds.right += extraSpaceEnd;
+                        }
+
+                        ensureTouchDelegateComposite(parent, view);
+                        ((TouchDelegateComposite) parent.getTouchDelegate())
+                                .addDelegate(bounds, view);
+                    }
+                }
+            });
+        }
+    }
+
+    private static void ensureTouchDelegateComposite(View parent, View view) {
+        TouchDelegate parentTouchDelegate = parent.getTouchDelegate();
+        if (parentTouchDelegate == null) {
+            parent.setTouchDelegate(new TouchDelegateComposite(new Rect(), view));
```

Comment:
do we often hit that case? I just wonder could we set the new bounds here instead of adding another delegate to a view in such cases?

Reply:
Usually a view won't have a `TouchDelegate` unless you give it one. Also the `new Rect(), view` part for `TouchDelegateComposite` is only there because it extends `TouchDelegate`. The two parameters aren't actually used.